numpy/core/src/umath/matmul.c.src

```diff
@@ -344,18 +356,25 @@ NPY_NO_EXPORT void
             }
         } else {
             /* matrix @ matrix */
-            if (i1blasable && i2blasable && o_ccontiguous) {
+            if (i1blasable && i2blasable && o_c_blasable) {
                 @TYPE@_matmul_matrixmatrix((void *)ip1, is1_m, is1_n,
                                            (void *)ip2, is2_n, is2_p,
-                                           (void *)op, dm, dn, dp);
+                                           (void *)op, os_m, os_p,
+                                           dm, dn, dp);
+            } else if (i1blasable && i2blasable && o_f_blasable) {
+                /*
+                 * Use transpose equivalence:
+                 * matmul(a, b, o) == matmul(b.T, a.T, o.T)
+                 */
+                @TYPE@_matmul_matrixmatrix((void *)ip2, is2_p, is2_n,
+                                           (void *)ip1, is1_n, is1_m,
+                                           (void *)op, os_p, os_m,
+                                           dm, dn, dp);
```

Comment:
I debated whether using `o_c_blasable`, `o_f_blasable` is clearer than pushing this all down into `@TYPE@_matmul_matrixmatrix`. I think this way is more explicit, even though the `os_m` parameter is not used at all. The other way would require flipping `ip1`, `ip2` and their strides inside the function, which would be messy.